src/main/resources/META-INF/rewrite/java-version-17.yml

```diff
@@ -122,3 +125,66 @@ recipeList:
   - org.openrewrite.java.ChangeMethodAccessLevel:
       methodPattern: "*..* premain(java.lang.String, java.lang.instrument.Instrumentation)"
       newAccessLevel: public
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.RemovedZipFinalizeMethods
+displayName: Remove `finalize` method in `java.util.zip.Zipfile`, `java.util.zip.Inflater` and `java.util.zip.Deflater`
+description: >
+  The `finalize` methods in `java.util.zip.ZipFile` is replaced with `close` method and is replaced by `end` method in 
+   `java.util.zip.Inflater` and `java.util.zip.Deflater` as it is no longer available in Java SE 12 and later.
+tags:
+  - java17
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "java.util.zip.Inflater finalize()"
+      newMethodName: end
+      ignoreDefinition: true
+      matchOverrides: true
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "java.util.zip.Deflater finalize()"
+      newMethodName: end
+      ignoreDefinition: true
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "java.util.ZipFile finalize()"
+      newMethodName: close
+      ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.RemovedSSLSessionGetPeerCertificateChainMethodImpl
+displayName: Remove `SSLSession.getPeerCertificateChain()` method
+description: >
+  The `javax.net.ssl.SSLSession.getPeerCertificateChain()` method implementation was removed from the SunJSSE provider and HTTP client implementation in Java SE 15. 
+  The default implementation will now throw an `UnsupportedOperationException`. 
+  Applications using this method should be updated to use the `javax.net.ssl.SSLSession.getPeerCertificates()` method instead.
+tags:
+  - java17
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "*.net.ssl.SSLSession getPeerCertificateChain()"
```

Comment:
Is there a reason we aren't using `javax` instead of `*`?

Reply:
During one test I had it as *, I reverted it back to javax now